scripts/setup-k8s-node.sh.in

```diff
@@ -34,14 +34,16 @@ HOSTNAME="$(hostname -f 2>/dev/null || curl http://169.254.169.254/latest/meta-d
 
 cat >/tmp/kubeadm-node.yaml <<EOF
 ---
-apiVersion: kubeadm.k8s.io/v1alpha2
-kind: NodeConfiguration
-token: "{{ClusterToken}}"
-discoveryFile: "/tmp/cluster-info.yaml"
+apiVersion: kubeadm.k8s.io/v1beta1
+kind: JoinConfiguration
+discovery:
+  tlsBootstrapToken: "{{ClusterToken}}"
+  file:
+    kubeConfigPath: "/tmp/cluster-info.yaml"
```

Comment:
This combination of tlsBootstrapToken and discovery file is working? I thought we hit issues with this in cluster-api and had to choose between one method or the other.

Reply:
Actually required this now: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go#L338-L339

Reply:
Ah, I was thinking of the mutually exclusive use of `bootstrapToken` and `file`. Good to know :)